MAP_ANONYMOUS undefined on older macOS

# syslog-ng
3.38.1

## Platform
macOS 10.10

# Issue
## Failure
Build fails thus:
```
lib/secret-storage/nondumpable-allocator.c:96:57: error: use of undeclared identifier 'MAP_ANONYMOUS'
  gpointer area = mmap(NULL, len, PROT_READ|PROT_WRITE, MAP_ANONYMOUS|MAP_PRIVATE, -1, 0);
                                                        ^
1 error generated.
```

## Input and output logs (if possible)
https://build.macports.org/builders/ports-10.10_x86_64-builder/builds/199959/steps/install-port/logs/stdio

## Discussion

`MAP_ANONYMOUS` used to be called `MAP_ANON` on older macOS versions, so the fix is simple:
```
#ifndef MAP_ANONYMOUS
#define MAP_ANONYMOUS MAP_ANON
#endif
```

Not sure if a compat header would be warranted when this is the only place it's used in the codebase.

--- a/lib/secret-storage/nondumpable-allocator.c
+++ b/lib/secret-storage/nondumpable-allocator.c
@@ -28,6 +28,11 @@
 #include <unistd.h>
 #include <stdio.h>
 
+/* For compatibility with e.g. older macOS. */
+#if !defined(MAP_ANONYMOUS) && defined(MAP_ANON)
+#define MAP_ANONYMOUS MAP_ANON
+#endif
+
 #include "nondumpable-allocator.h"
 
 #define ALLOCATION_HEADER_SIZE offsetof(Allocation, user_data)

